### PR TITLE
Fix infinite Loader on Carousel when slide breaks

### PR DESCRIPTION
**Pull request by tkovs**
#### What is the purpose of this pull request?
Fix the infinite Loader when a Carousel slide has problems

#### How should this be manually tested?
By adding wrong urls to `Carousel/index.mdx` and check the Carousel page on docz locally.

#### Types of changes
- [x] Refactor (non-breaking change which improves security, performance or maintainability)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Carousel/index.js [hunk 1]
```diff
@@ -21,6 +22,15 @@ const initialState = {
   alt: [],
   thumbsLoaded: false,
   activeIndex: 0,
+  failedImageIndexes: [],
```
Comment (tkovs) [inline react/components/Carousel/index.js:25]: Basicamente eu anoto a posição dos slides que deram problema, e na hora de exibir os slides eu mostro o placeholder caso o index do slide tenha sido anotado anteriormente

File: react/components/Carousel/index.js [hunk 2]
```diff
@@ -41,7 +51,7 @@ class Carousel extends Component {
     this.thumbLoadCount = 0
 
     if (slides) {
-      const thumbPromises = slides.flatMap((slide, i) => {
+      const thumbPromises = slides.map((slide, i) => {
```
Comment (tkovs) [inline react/components/Carousel/index.js:54]: Antes era usado um flatMap simulando um map com filter, pois ignorava videos que davam problema. Agora até mesmo um video com problema mostra placeholder.

File: react/components/Video/Vimeo.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ class Vimeo extends Component {
         .then(response => {
           resolve(Vimeo.thumbUrlFromResp(response, thumbWidth))
         })
+        .catch(() => reject())
```
Comment (tkovs) [inline react/components/Video/Vimeo.js:15]: Não estava conseguindo capturar o catch daqui na promise thumbUrl lá no componente Carousel, então deixei explicito assim.

**Review by Lurian — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Fixed
+- Carousel slide error runs into an infinite Loader
```
Comment (Lurian) [inline CHANGELOG.md]:
```suggestion
- Carousel slide error running into an infinite loading state
```

File: react/components/Carousel/index.js [hunk 5]
```diff
@@ -195,7 +226,7 @@ class Carousel extends Component {
   render() {
     const { thumbSwiper, thumbsLoaded } = this.state
     const { rebuildGalleryOnUpdate } = this
-    const { slides } = this.props
+    const slides = this.props.slides.map(slide => ({ ...slide }))
```
Comment (Lurian) [inline react/components/Carousel/index.js]:
```suggestion
    const slides = this.props.slides ? this.props.slides.map(slide => { ...slide }) : []
```

**Review by Lurian — CHANGES_REQUESTED** [review]

**Review by Lurian — APPROVED** [review]

**Pull request merged**